src/lib.rs

```diff
@@ -158,6 +162,16 @@ impl Pty {
         // and it's not safe to keep it open
         libc::close(raw_controller_fd);
 
+        // just to be safe, mark every single file descriptor as close-on-exec.
```

Comment:
why do we need to set this in addition to O_CLOEXEC on controller and user fds? isnt the latter sufficient?

Reply:
defense in depth. (you could argue that you _do_ need the `O_CLOEXEC` for macos, since this is not available there).